src/synth/synth-expr.adb

```diff
@@ -1891,6 +1891,42 @@ package body Synth.Expr is
       return Create_Value_Net (Dff, Expr.Typ);
    end Synth_Psl_Prev;
 
+   function Synth_Psl_Stable (Syn_Inst : Synth_Instance_Acc; Call : Node)
+                              return Valtyp
+   is
+      Ctxt      : constant Context_Acc := Get_Build (Syn_Inst);
+      Clock     : Node;
+      Dff       : Net;
+      Clk       : Valtyp;
+      Expr      : Valtyp;
+      Clk_Net   : Net;
+      CurVal    : Valtyp;
+      OldVal    : Valtyp;
+      Res       : Valtyp;
+   begin
+      Expr := Synth_Expression (Syn_Inst, Get_Expression (Call));
+
+      Clock := Get_Clock_Expression (Call);
+      if Clock /= Null_Node then
+         Clk := Synth_Expression (Syn_Inst, Clock);
+         Clk_Net := Get_Net (Ctxt, Clk);
+      else
+         Clock := Get_Default_Clock (Call);
+         pragma Assert (Clock /= Null_Node);
+         Clk_Net := Synth_PSL_Expression (Syn_Inst, Get_Psl_Boolean (Clock));
+      end if;
```

Comment:
At this point, the clock is also synthesized.  This code block should be factorized with the one for prev().

Reply:
Factorized into separate function `Synth_Psl_Function_Clock()`